src/hivpy/hiv_status.py

```diff
@@ -390,6 +391,8 @@ def disease_and_diagnosis(disease_col, diagnosis_col, disease_rate, diagnosis_pr
         pop.set_present_variable(col.WHO3_EVENT, (who3_disease | tb), HIV_pos)
         new_tb = pop.get_sub_pop_from_array(tb, HIV_pos)
         pop.set_present_variable(col.TB_INFECTION_DATE, pop.date, new_tb)
+        pop.set_present_variable(col.TB_PRIMARY_INFECTION, False)  # FIXME: move this reset elsewhere?
+        pop.set_present_variable(col.TB_PRIMARY_INFECTION, True, new_tb)
```

Comment:
There should be somewhere better to put the reset of TB primary infection status; thoughts?